rusk/src/lib/http/event.rs

```diff
@@ -97,84 +104,103 @@ impl MessageRequest {
     }
 }
 
-#[derive(Debug, Deserialize, Serialize, Default)]
+#[derive(Debug)]
 pub(crate) struct MessageResponse {
     pub headers: serde_json::Map<String, serde_json::Value>,
 
     /// The data returned by the contract call.
-    pub data: DataType,
+    pub data: ResponseData,
 
     /// A possible error happening during the contract call.
-    #[serde(skip_serializing_if = "Option::is_none")]
     pub error: Option<String>,
 }
 
 impl MessageResponse {
     pub fn from_error(error: String) -> Self {
         Self {
+            headers: serde_json::Map::default(),
+            data: ResponseData::None,
             error: Some(error),
-            ..Default::default()
         }
     }
 
-    pub fn to_bytes(&self) -> anyhow::Result<Vec<u8>> {
-        Ok(vec![])
-    }
-
-    pub fn to_http(
-        &self,
-        binary: bool,
-    ) -> anyhow::Result<hyper::Response<hyper::Body>> {
+    pub fn into_http(self) -> anyhow::Result<hyper::Response<hyper::Body>> {
         if let Some(error) = &self.error {
             return Ok(hyper::Response::builder()
                 .status(hyper::StatusCode::INTERNAL_SERVER_ERROR)
                 .body(hyper::Body::from(error.to_string()))?);
         }
 
-        let body = match binary {
-            true => self.to_bytes()?,
-            false => match &self.data {
-                DataType::Binary(BinaryWrapper { inner }) => hex::encode(inner),
-                DataType::Text(text) => text.to_string(),
-                DataType::None => String::default(),
+        let body = {
+            match self.data {
+                ResponseData::Binary(wrapper) => Body::from(wrapper.inner),
```

Comment:
This should check the original application/content.

It can happen that the client doesn't understand binary, hence it send binary in form of hex encoding

```suggestion
                ResponseData::Binary(wrapper) => {
                    let data = match is_binary {
                        true => wrapper.inner,
                        false => hex::encode(wrapper.inner).as_bytes().to_vec(),
                    };
                    Body::from(data)
                }

```


To test it try to send the following request to http://127.0.0.1:8080/01/0100000000000000000000000000000000000000000000000000000000000000

```json
{
    "topic": "module_balance",
    "data": "0200000000000000000000000000000000000000000000000000000000000000"
}
```

The result shall be `0010a5d4e8000000` and not raw_binary